lib/srv/db/server.go

```diff
@@ -326,6 +344,65 @@ func New(ctx context.Context, config Config) (*Server, error) {
 	return server, nil
 }
 
+func (s *Server) startSQLServerLDAP(ctx context.Context, database types.Database) error {
+	clusterName, err := s.cfg.AuthClient.GetClusterName()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	caCert, err := x509.ParseCertificate([]byte(database.GetAD().LDAPCert))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	ad := database.GetAD()
+	serverName := ad.KDCHostName
+
+	if ad.ServerName != "" {
+		serverName = ad.ServerName
+	}
+
+	s.caClients[database.GetName()] = windows.NewCertificateStoreClient(windows.CertificateStoreConfig{
```

Comment:
Does it make sense to use domains as a key  instead of database name  and detect the situation when the initializeLDAP is already running for particular domain ?